src/pages/pageHome/modal/addNewCat.tsx

```diff
@@ -0,0 +1,47 @@
+import { Cat } from 'components/main/main';
+import React from 'react';
+import { Form } from '../form/form';
+import classes from './addNewCat.module.css';
+
+type PropsFunc = {
+  createCat: (cat: Cat) => void;
+  toggleModal: () => void;
+};
+
+export class AddNewCat extends React.Component<PropsFunc> {
+  handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    const form = e.target as HTMLFormElement;
+    const formData = Object.fromEntries(new FormData(form).entries());
+
+    const randomImage = async () => {
+      return await fetch('https://source.unsplash.com/random?cat').then((data) => data.url);
+    };
```

Comment:
Разве так не сработает?
```suggestion
    const randomImage = await fetch('https://source.unsplash.com/random?cat').then((data) => data.url);
```